lib/mapObjects/IMarket.h

```diff
@@ -36,6 +36,11 @@ class DLL_LINKAGE IMarket : public virtual Serializeable, boost::noncopyable
 	CArtifactSet * getArtifactsStorage() const;
 	bool getOffer(int id1, int id2, int &val1, int &val2, EMarketMode mode) const; //val1 - how many units of id1 player has to give to receive val2 units
 
+	template <typename Handler> void serializeArtifactsAltar(Handler &h)
+	{
+		h & *altarArtifactsStorage;
```

Comment:
why should we save artifacts storage?

Reply:
Because we did so in 1.5 saves.

This code will never be called when saving new games in 1.6 or when loading 1.6 save and will be removed once we drop support for 1.5 saves.